OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -191,30 +221,33 @@ public override Activity Tick(Actor self)
 			if (firstFacing != mobile.Facing)
 			{
 				path.Add(nextCell.Value.First);
-				return ActivityUtils.SequenceActivities(new Turn(self, firstFacing), this);
+				QueueChild(new Turn(self, firstFacing));
+				return this;
 			}
-			else
-			{
-				mobile.SetLocation(mobile.FromCell, mobile.FromSubCell, nextCell.Value.First, nextCell.Value.Second);
-
-				var map = self.World.Map;
-				var from = (mobile.FromCell.Layer == 0 ? map.CenterOfCell(mobile.FromCell) :
-					self.World.GetCustomMovementLayers()[mobile.FromCell.Layer].CenterOfCell(mobile.FromCell)) +
-					map.Grid.OffsetOfSubCell(mobile.FromSubCell);
 
-				var to = Util.BetweenCells(self.World, mobile.FromCell, mobile.ToCell) +
-					(map.Grid.OffsetOfSubCell(mobile.FromSubCell) + map.Grid.OffsetOfSubCell(mobile.ToSubCell)) / 2;
-
-				var move = new MoveFirstHalf(
-					this,
-					from,
-					to,
-					mobile.Facing,
-					mobile.Facing,
-					0);
-
-				return move;
-			}
+			mobile.SetLocation(mobile.FromCell, mobile.FromSubCell, nextCell.Value.First, nextCell.Value.Second);
+
+			var map = self.World.Map;
+			var from = (mobile.FromCell.Layer == 0 ? map.CenterOfCell(mobile.FromCell) :
+				self.World.GetCustomMovementLayers()[mobile.FromCell.Layer].CenterOfCell(mobile.FromCell)) +
+				map.Grid.OffsetOfSubCell(mobile.FromSubCell);
+
+			var to = Util.BetweenCells(self.World, mobile.FromCell, mobile.ToCell) +
+				(map.Grid.OffsetOfSubCell(mobile.FromSubCell) + map.Grid.OffsetOfSubCell(mobile.ToSubCell)) / 2;
+
+			QueueChild(new MoveFirstHalf(
+				this,
+				from,
+				to,
+				mobile.Facing,
+				mobile.Facing,
+				0));
+
+			// While carrying out one Move order, MoveSecondHalf finishes its work from time to time and returns null.
+			// That causes the ChildActivity to be null and makes us return to this part of this function.
+			// If we only queue the activity and not run it, units will pause for one tick!
+			ChildActivity = ActivityUtils.RunActivity(self, ChildActivity);
```

Comment:
The problem might be with

https://github.com/forcecore/OpenRA/blob/ab536f5452da590ace46797e4bf22dd7753cc6df/OpenRA.Mods.Common/Activities/Move/Move.cs#L218

If the path becomes invalid then it will stop and wait for the next tick before attempting to repath.